checks/evaluation/cii_best_practices.go

```diff
@@ -15,48 +15,93 @@
 	// https://github.com/ossf/scorecard-action/blob/main/policies/template.yml#L61.
 	passingScore    = 5
 	inProgressScore = 2
+	maxScore        = 10
+	minScore        = 0
+	errScore        = -1
 )
 
 // CIIBestPractices applies the score policy for the CIIBestPractices check.
-func CIIBestPractices(name string, _ checker.DetailLogger, r *checker.CIIBestPracticesData) checker.CheckResult {
-	if r == nil {
-		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
+func CIIBestPractices(name string,
+	findings []finding.Finding, dl checker.DetailLogger,
+) checker.CheckResult {
+	expectedProbes := []string{
+		hasGoldBadge.Probe,
+		hasSilverBadge.Probe,
+		hasInProgressBadge.Probe,
+		hasPassingBadge.Probe,
+		hasUnknownBadge.Probe,
+	}
+
+	if !finding.UniqueProbesEqual(findings, expectedProbes) {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "invalid probe results")
+		return checker.CreateRuntimeErrorResult(name, e)
+	}
+
+	var countBadges int
+	// Sanity check that we don't have multiple positives
+	for i := range findings {
+		f := &findings[i]
+
+		if f.Outcome == finding.OutcomePositive {
+			countBadges++
+		}
+	}
+
+	var score int
+	var text string
+
+	if countBadges > 1 {
+		errText := "invalid probe results: multiple badges detected"
+		e := sce.WithMessage(sce.ErrScorecardInternal, errText)
 		return checker.CreateRuntimeErrorResult(name, e)
+	} else if countBadges == 0 {
+		text = "no effort to earn an OpenSSF best practices badge detected"
+		return checker.CreateResultWithScore(name, text, minScore)
 	}
 
-	var results checker.CheckResult
-	switch r.Badge {
-	case clients.NotFound:
-		results = checker.CreateMinScoreResult(name, "no effort to earn an OpenSSF best practices badge detected")
-	case clients.InProgress:
-		msg := fmt.Sprintf("badge detected: %v", r.Badge)
-		results = checker.CreateResultWithScore(name, msg, inProgressScore)
-	case clients.Passing:
-		msg := fmt.Sprintf("badge detected: %v", r.Badge)
-		results = checker.CreateResultWithScore(name, msg, passingScore)
-	case clients.Silver:
-		msg := fmt.Sprintf("badge detected: %v", r.Badge)
-		results = checker.CreateResultWithScore(name, msg, silverScore)
-	case clients.Gold:
-		msg := fmt.Sprintf("badge detected: %v", r.Badge)
-		results = checker.CreateMaxScoreResult(name, msg)
-	case clients.Unknown:
-		e := sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("unsupported badge: %v", r.Badge))
-		results = checker.CreateRuntimeErrorResult(name, e)
+	for i := range findings {
+		f := &findings[i]
+		if f.Outcome == finding.OutcomePositive {
+			switch f.Probe {
+			case hasInProgressBadge.Probe:
+				score = inProgressScore
+				text = "badge detected: InProgress"
+			case hasPassingBadge.Probe:
+				score = passingScore
+				text = "badge detected: Passing"
+			case hasSilverBadge.Probe:
+				score = silverScore
+				text = "badge detected: Silver"
+			case hasGoldBadge.Probe:
+				score = maxScore
```

Comment:
I can understand why you have a const declared for this case, instead of using `checker.CreateMaxScoreResult`, since you're trying to de-duplicate the logic. But at least switch this to use `checker.MaxResultScore` instead of the locally declared constant.
 https://github.com/ossf/scorecard/blob/6dffe650007a62be5b357f009581370621b55872/checker/check_result.go#L34